windows/src/global/delphi/packages/Keyman.System.PackageInfoRefreshKeyboards.pas

```diff
@@ -77,9 +109,48 @@ procedure TPackageInfoRefreshKeyboards.Execute;
         pack.Keyboards.Add(k);
       end;
 
-      FillKeyboardDetails(pack.Files[i], k);
+      if FillKeyboardDetails(pack.Files[i], pki) then
+      begin
+        k.Name := pki.Name;
+        k.ID := pki.ID;
+        k.Version := pki.Version;
+      end;
+    end;
+  end;
+
+  // Test that each keyboard has at most one target file for each platform and
+  // that the versions match
+  for i := 0 to pack.Keyboards.Count - 1 do
+  begin
+    ids := FindKeyboardFilesByID(pack.Keyboards[i].ID);
+    Assert(Length(ids) > 0);
+
+    if Length(ids) = 1 then
+      Continue;
+
+    if Length(ids) > 2 then
+    begin
+      DoError(Format(SError_TooManyTargetFilesForKeyboardID, [pack.Keyboards[i].ID, Length(ids)]), plsError);
+      Exit(False);
```

Comment:
I assume, based on this logic, that this is omitting the possibility for a LDML keyboard at this time.  Which is reasonable; I'm just wanting to ensure I understand what this block and the one immediately after it are doing.

Reply:
Yes, we'll add the LDML logic later :)